gapis/api/watcher.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/gapis/memory"
+)
+
+type Field interface {
+	FieldName() string
+	ClassName() string
+}
+
+// StateWatcher provides callbacks to track state effects
+type StateWatcher interface {
+	// OnBeginCmd is called at the beginning of each API call
+	OnBeginCmd(ctx context.Context, cmdID CmdID, cmd Cmd)
+
+	// OnEndCmd is called by the API emulator at the end of each API call
```

Comment:
The term used for command 'emulation' I'm sticking with is 'executor'.
Regardless, maybe match the documentation of `OnBeginCmd`:

`// OnEndCmd is called at the end of each API call`

Reply:
Done